src/System.Management.Automation/engine/ExperimentalFeature/GetExperimentalFeatureCommand.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Internal;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// Implements Get-ExperimentalFeature cmdlet.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "ExperimentalFeature", HelpUri = "")]
+    public class GetExperimentalFeatureCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Get and set the feature names.
+        /// </summary>
+        [Parameter(ValueFromPipeline = true, Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string[] Name { get; set; }
+
+        /// <summary>
+        /// Get and set the switch flag to search module paths to find all available experimental features.
+        /// </summary>
+        [Parameter]
+        public SwitchParameter ListAvailable { get; set; }
+
+        /// <summary>
+        /// ProcessRecord method of this cmdlet.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            const WildcardOptions wildcardOptions = WildcardOptions.IgnoreCase | WildcardOptions.CultureInvariant;
+            IEnumerable<WildcardPattern> namePatterns = SessionStateUtilities.CreateWildcardsFromStrings(Name, wildcardOptions);
+
+            if (ListAvailable)
+            {
+                foreach (ExperimentalFeature feature in GetAvailableExperimentalFeatures(namePatterns).OrderBy(GetSortingString))
+                {
+                    WriteObject(feature);
+                }
+            }
+            else if (ExperimentalFeature.EnabledExperimentalFeatureNames.Count > 0)
+            {
+                foreach (ExperimentalFeature feature in GetEnabledExperimentalFeatures(namePatterns).OrderBy(GetSortingString))
+                {
+                    WriteObject(feature);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Construct the string for sorting experimental feature records.
+        /// </summary>
+        /// <remarks>
+        /// Engine features come before module features.
+        /// Within engine features and more features, features are ordered by name.
+        /// </remarks>
+        private static string GetSortingString(ExperimentalFeature feature)
+        {
+            if (ExperimentalFeature.EngineSource.Equals(feature.Source))
```

Comment:
Does this need to be case insensitive?

Reply:
I was assuming people will use the const string field `ExperimentalFeature.EngineSource` for registering an engine exp feature, but it's more reliable to make this comparison case insensitive.